### PR TITLE
[WIP] tests: Migrate legacy test 24

### PR DESCRIPTION
**Pull request by AitorATuin**

### CODE REVIEW

File: test/functional/legacy/024_regexp_with_backslash_spec.lua [hunk 1]
```diff
@@ -0,0 +1,158 @@
+-- Tests for regexp with backslash and other special characters inside []
+-- Also test backslash for hex/octal numbered character.
+--
+-- Note:
+-- Since this text input contains null characters (\x00) we can't pass the
+-- full input to `insert`. We need instead to insert the input reading 
+-- from 3 files and insert the null bytest manually between them.
```
Comment (brcolow) [inline test/functional/legacy/024_regexp_with_backslash_spec.lua]: bytest -> bytes

Comment (AitorATuin) [inline test/functional/legacy/024_regexp_with_backslash_spec.lua]: ops, fixing it :+1:

**Comment by justinmk** [comment]
@AitorATuin please rebase, then will merge.

**Comment by AitorATuin** [comment]
> please rebase, then will merge.

Sure, I will do it

**Comment by AitorATuin** [comment]
@justinmk rebased!

**Comment by justinmk** [comment]
test24 was removed upstream and we merged that.

**Pull request closed without merging**